internal/state/ownership/tree.go

```diff
@@ -180,3 +184,48 @@ func headRoot(contract common.Address, store storage.Tx) (common.Hash, error) {
 func setHeadRoot(contract common.Address, store storage.Tx, root common.Hash) error {
 	return store.Set([]byte(headRootKeyPrefix+contract.String()), root.Bytes())
 }
+
+// TagRoot stores a root value for the block so that it can be checked later
+func (b *tree) TagRoot(blockNumber int64) error {
+	tagKey := tagPrefix + b.contract.String() + "/" + strconv.FormatInt(blockNumber, 10)
+	root := b.Root()
+	return b.store.Set([]byte(tagKey), root.Bytes())
+}
+
+// Checkout sets the current root to the one that is tagged for a blockNumber.
+func (b *tree) Checkout(blockNumber int64) error {
+	tagKey := tagPrefix + b.contract.String() + "/" + strconv.FormatInt(blockNumber, 10)
+	buf, err := b.store.Get([]byte(tagKey))
+	if err != nil {
+		return err
+	}
+
+	if len(buf) == 0 {
+		return errors.New("no tag found for this block number " + strconv.FormatInt(blockNumber, 10))
+	}
+
+	newRoot := common.BytesToHash(buf)
+	b.mt.SetRoot(newRoot)
+	return setHeadRoot(b.contract, b.store, newRoot)
```

Comment:
Can you write a TODO here where you explain that we have to rethink the approach as we might accidentally call Commit instead of Discard after calling this method, please?

Reply:
I added comments related to this in the Checkout function in the state file